x-pack/plugins/search_playground/public/plugin.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CoreSetup,
+  Plugin,
+  CoreStart,
+  AppMountParameters,
+  DEFAULT_APP_CATEGORIES,
+} from '@kbn/core/public';
+import { PLUGIN_NAME } from '../common';
+import { PlaygroundToolbar, PlaygroundProvider, Playground } from './embedable';
+import {
+  AppPluginStartDependencies,
+  SearchPlaygroundPluginSetup,
+  SearchPlaygroundPluginStart,
+} from './types';
+
+export class SearchPlaygroundPlugin
+  implements Plugin<SearchPlaygroundPluginSetup, SearchPlaygroundPluginStart>
+{
+  public setup(core: CoreSetup): SearchPlaygroundPluginSetup {
+    core.application.register({
```

Comment:
This should be gated with a config value, but we can add that in a follow-up.

Reply:
Do you have a link to a file with that?

Reply:
we do it in [enterprise_search](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/public/plugin.ts#L142-L145)